### ISSUE TITLE
Spacemacs can't create auto-save files when using tramp on Windows.

### ISSUE THREAD

**Issue by batonius** [syl20bnr/spacemacs#7302]
#### Description :octocat:

Spacemacs can't create auto-save files in the cache directory when using tramp on Windows. `dotspacemacs-auto-save-file-location` is set to `'cache`.
#### Reproduction guide :beetle:
- Start Emacs on Windows via msys2.
- Open a file on a remote server with tramp.
- Edit the file.
- Wait for auto-save to kick in.

_Observed behaviour:_ :eyes: :broken_heart:
No new auto-save files in .cache/dist or .cache/site directories, error message in the Messages buffer:

```
Auto-saving test.sh: Opening output file: Invalid argument, d:/msys64/home/user/.emacs.d/.cache/auto-save/dist/#!plinkx:188:!home!user!test.sh#
```

_Expected behaviour:_ :heart: :smile:
Auto-saving working properly. I've looked in `~/.emacs.d/.cache/auto-save` and found auto-save files created by the previous version of Spacemacs in the `site` subdir, their names being of form `#_aplinkx_b188_b_ahome_auser_atest.sh#`, so I guess the problem is broken file name escaping.
#### System Info :computer:
- OS: windows-nt
- Emacs: 25.1.1
- Spacemacs: 0.200.0
- Spacemacs branch: master (rev. e6357a5)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:

``` elisp
(helm emacs-lisp dockerfile
      (shell :variables shell-default-height 30 shell-default-shell 'shell shell-default-position 'bottom)
      rust markdown c-c  
      (colors :variables colors-colorize-identifiers 'all)
      sql php html)
```

**Comment by TheBB** [comment]
If Spacemacs caused this to happen it should be possible to bisect this. If not, you should probably report this upstream to the relevant places, since I don't *think* (famous last words) we are doing anything that might affect this.

**Comment by batonius** [comment]
You're right, the problem is with vanilla emacs 25.1.

 I've looked into it and it looks like tramp has two methods for auto-save file name escaping: if `tramp-auto-save-directory` is set it uses it's own transformations (/ -> _a, : -> _b, etc) and if it's nil it uses the default behavior for the system provided by `make-auto-save-file-name`. The problem is, after [this commit](http://git.savannah.gnu.org/cgit/tramp.git/commit/lisp/tramp.el?id=7722b7cfabf667e774cdfbb56c25b2bb9eebe9fb) `make-auto-save-file-name` can't detect the system as windows-nt and fails to replace the forbidden symbols in file names. I don't understand the reasoning behind the commit, but I got auto-saving working by explicitly setting `tramp-auto-save-directory`:
```
 (setq tramp-auto-save-directory (concat spacemacs-auto-save-directory "tramp/"))
```

I'm aware this isn't your bug, but maybe Spacemacs could set `tramp-auto-save-directory` when it sets `auto-save-file-name-transforms`? It seems reasonable and would solve the problem.
